lib/active_record/immutable.rb

```diff
@@ -0,0 +1,49 @@
+module ActiveRecord
+  module Immutable
+    extend ActiveSupport::Concern
+
+    FIELD_NAME_IDENTIFIER_IMMUTABLE = 'previous_version_id'.freeze
+
+    included do
+      def self.default_scope
+        where(%{
+          not exists (
+            select 1
+            from #{quoted_table_name} model
+            where model.#{FIELD_NAME_IDENTIFIER_IMMUTABLE} = #{quoted_table_name}.id
+          )
+        })
+      end
+
+      def _update_record(_attribute_names = self.attribute_names)
```

Comment:
Where will we be using this method? And why does it start with an underscore?

Reply:
This is a active_record method to update the record, so I'm overwritten this to not update the record.

I can put a comment to describe this. What do you think?